src/testing.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This module provides a set of utility functions that are useful for unit testing Cloud Functions.
+import {Server} from 'http';
+
+import {CloudEventsContext, HandlerFunction} from '.';
+import {getRegisteredFunction} from './function_registry';
+import {getServer} from './server';
+
+/**
+ * The default CloudEvent returned by testCloudEvent
+ */
+const TEST_CLOUD_EVENT: CloudEventsContext = {
+  specversion: '1.0',
+  type: 'com.google.cloud.functions.test',
+  source: 'https://github.com/GoogleCloudPlatform/functions-framework-nodejs',
+  id: 'aaaaaa-1111-bbbb-2222-cccccccccccc',
+  time: '2020-05-13T01:23:45Z',
+  datacontenttype: 'application/json',
+  data: {
+    test: 'payload',
+  },
+};
+
+/**
+ * Testing utility for retrieving a function registered with the Functions Framework
+ * @param functionName the name of the function to get
+ * @returns a function that was registered with the Functions Framework
+ *
+ * @beta
+ */
+export const getFunction = (
+  functionName: string
+): HandlerFunction | undefined => {
+  return getRegisteredFunction(functionName)?.userFunction;
+};
```

Comment:
Discussion: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/pull/392#discussion_r755617031

I believe it would be reasonable to change the location of the export. Specifically, export `getFunction` with the other function registry exports here:
https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/src/index.ts

That's my suggestion.

---

I still believe this useful function is not necessarily tied to testing and am suggesting we simply move the export location. There is no testing logic in this function and the same `@beta` tag can still be provided there.

We shouldn't mix import locations for testing and registry functions.

Reply:
My strong preference is to not expose this function from the primary API in `index.ts`. My primary concern is that I expect some flux in this API as we add more functionality to the framework (e.g. lifecycle events). Exporting from here sends a strong signal to developers that it's only purpose is for testing and it is not to be used in production.

Reply:
This is fine. I don't agree that the location of the export will change developer behavior (see recent internal code location changes for example), but the difference is insignificant.